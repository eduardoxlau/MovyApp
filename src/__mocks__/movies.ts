const movies = [
  {
    id: 143,
    title: 'International Research Orchestrator',
    overview:
      'Blanditiis similique pariatur provident dolorem repellendus. Accusamus numquam alias in saepe pariatur tenetur. Dolor consequatur quae repudiandae nulla ullam iusto odit repellat qui. Impedit sit facere nostrum nobis dolor. Magnam maiores sunt adipisci excepturi nulla sed aperiam.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view1.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 4,
    genres: [
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
  {
    id: 140,
    title: 'International Brand Representative',
    overview:
      'Iure vel aut exercitationem possimus neque animi. Quo aperiam molestiae consectetur. Soluta exercitationem rem voluptate ipsa ipsum quisquam eum. Aut qui accusamus minus nostrum excepturi et tempore ex eos. Culpa eum perferendis laboriosam ea.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view1.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 2,
    genres: [
      {
        name: 'Fun',
      },
    ],
  },
  {
    id: 147,
    title: 'Future Response Designer',
    overview:
      'Doloribus ducimus quia ex qui quia tenetur aperiam odit ut. Amet repellendus laudantium. Omnis adipisci est ut dolor doloremque et. Sunt consequatur dolorem et dolore rem qui. Voluptatem necessitatibus aperiam.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view1.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 5,
    genres: [
      {
        name: 'Action',
      },
    ],
  },
  {
    id: 134,
    title: 'Future Data Consultant',
    overview:
      'Distinctio accusamus enim optio quis. Dignissimos non autem. Dolores fugit architecto quo aut.',
    card_url: 'http://localhost:4000/card1.png',
    poster_url: 'http://localhost:4000/view4.png',
    backdrop_url: 'http://localhost:4000/avatar.png',
    stars: 3,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
  {
    id: 149,
    title: 'Forward Tactics Administrator',
    overview:
      'Ipsam officia et illo. Sit aut illo consequatur rerum ipsum culpa temporibus et laboriosam. Cupiditate aut ipsam modi dolor facilis fugit maiores. Recusandae veniam ad doloremque corrupti fuga molestiae sit placeat. Sint quisquam illo consequatur itaque.',
    card_url: 'http://localhost:4000/more2.png',
    poster_url: 'http://localhost:4000/view2.png',
    backdrop_url: 'http://localhost:4000/matrix.png',
    stars: 1,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
  {
    id: 139,
    title: 'Forward Solutions Liaison',
    overview:
      'Quasi est hic ipsa ipsa quod. Incidunt eius maiores aut. Ut nihil molestiae harum exercitationem nisi. Esse repellat pariatur hic porro deleniti iste sint. Sapiente non iste dolores nulla.',
    card_url: 'http://localhost:4000/more4.png',
    poster_url: 'http://localhost:4000/view3.png',
    backdrop_url: 'http://localhost:4000/padrino.jpeg',
    stars: 3,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
    ],
  },
  {
    id: 144,
    title: 'Dynamic Group Representative',
    overview:
      'Optio assumenda ipsum quos temporibus repudiandae et. Esse doloribus voluptas. Error molestiae error.',
    card_url: 'http://localhost:4000/more4.png',
    poster_url: 'http://localhost:4000/view3.png',
    backdrop_url: 'http://localhost:4000/padrino.jpeg',
    stars: 1,
    genres: [
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
    ],
  },
  {
    id: 142,
    title: 'Dynamic Communications Executive',
    overview:
      'Tempora incidunt et. Officia blanditiis dolorem ex. Est vitae commodi error culpa quas doloremque. Quo dolores suscipit ex velit et tempora.',
    card_url: 'http://localhost:4000/card2.png',
    poster_url: 'http://localhost:4000/view2.png',
    backdrop_url: 'http://localhost:4000/matrix.png',
    stars: 2,
    genres: [
      {
        name: 'Action',
      },
      {
        name: 'Fun',
      },
      {
        name: 'Horror',
      },
      {
        name: 'Family',
      },
    ],
  },
];

export default movies;
