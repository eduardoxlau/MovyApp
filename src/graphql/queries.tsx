import { gql } from '@apollo/client';

export const GET_USER = gql`
  query GetUser {
    getUser {
      id
      email
      full_name
      admin
      lists {
        description
        movies {
          id
          title
        }
      }
    }
  }
`;

export const GET_MOVIES_SEEN = gql`
  query GetMoviesSeen {
    getMoviesSeen {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIE = gql`
  query GetMovie($id: Float!) {
    getMovie(id: $id) {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIES = gql`
  query GetMovies {
    getMovies(limit: 20) {
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_MOVIES_PAGINATE = gql`
  query GetMovies($page: Float, $limit: Float) {
    getMovies(page: $page, limit: $limit) {
      meta {
        totalPages
        currentPage
      }
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_LISTS = gql`
  query GetLists {
    getLists {
      id
      name
      description
      movies {
        id
        title
        trailer_url
        card_url
      }
    }
  }
`;
