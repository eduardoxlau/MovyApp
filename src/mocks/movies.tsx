import { ItemInterface } from 'components/card';

import Card1 from 'assets/card1.png';
import Card2 from 'assets/card2.png';
import Card3 from 'assets/card3.png';

import View1 from 'assets/view1.png';
import View2 from 'assets/view2.png';
import View3 from 'assets/view3.png';
import View4 from 'assets/view4.png';

import More1 from 'assets/more1.png';
import More2 from 'assets/more2.png';
import More3 from 'assets/more3.png';
import More4 from 'assets/more4.png';

export const moviesSuggestions: ItemInterface[] = [
  {
    id: 0,
    img: More1,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
  {
    id: 1,
    img: More2,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
  {
    id: 2,
    img: More3,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
  {
    id: 3,
    img: More4,
    title: 'The matrix',
    description:
      'It depicts a dystopian "future in which reality as perceived by humans is actually" a simulated reality called the Matrix.',
  },
];

export const movies: ItemInterface[] = [
  {
    id: 0,
    title: 'The matrix',
    img: Card2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 1,
    title: 'The Shawshank Redemption ',
    img: Card3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 2,
    title: 'El Padrino',
    img: Card1,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 3,
    title: 'The matrix',
    img: Card2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 4,
    title: 'The Shawshank Redemption ',
    img: Card3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 5,
    title: 'El Padrino',
    img: Card1,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
];

export const moviesByMovy: ItemInterface[] = [
  {
    id: 0,
    title: 'The Dark Knight',
    img: View1,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 1,
    title: '12 Angry Men',
    img: View2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 2,
    title: 'La lista de Schindler',
    img: View3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 3,
    title: ' Pulp Fiction',
    img: View4,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 4,
    title: '12 Angry Men',
    img: View2,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 5,
    title: 'La lista de Schindler',
    img: View3,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
  {
    id: 6,
    title: ' Pulp Fiction',
    img: View4,
    description:
      'During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a fierce storm and left behind by his crew.',
  },
];
