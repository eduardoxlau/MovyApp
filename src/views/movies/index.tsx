import { useState } from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { useQuery } from '@apollo/client';

import Loading from 'components/loading';
import { ItemInterface } from 'components/card';
import { GET_MOVIES_PAGINATE } from 'graphql/queries';

const Movies = () => {
  const [page, setPage] = useState(1);
  const { data, loading } = useQuery(GET_MOVIES_PAGINATE, {
    variables: {
      page,
      limit: 8,
    },
  });

  if (loading) return <Loading />;

  const pages = data.getMovies.meta.totalPages;
  const movies = data.getMovies.items;

  return (
    <div className="bg-black flex flex-col flex-grow">
      <div className="container mx-auto mt-40 mb-20">
        <div className="flex w-full flex-col md:flex-row text-white flex-wrap">
          {movies?.map((movie: ItemInterface) => (
            <div
              className="flex flex-col px-3 w-full md:w-1/4 mb-10 cursor-pointer"
              key={movie.id}
            >
              <Link to={`/trailer/${movie.id}`}>
                <img className="h-50 w-full" src={movie.card_url} alt="" />
                <div className="flex my-4 items-center">
                  <div className="text-xl">{movie.title}</div>
                  <div className="rounded-full border border-white ml-auto text-lg px-4 py-0 font-bold opacity-90">
                    PG13
                  </div>
                </div>
                <div>{`${movie.overview.slice(0, 100)} ...`}</div>
              </Link>
            </div>
          ))}
        </div>
        <div className="flex justify-center text-center text-white">
          {Array.from(Array(pages).keys()).map((_, index) => (
            <div
              className={classNames({
                'm-1 px-2 py-1 rounded cursor-pointer': true,
                'bg-blue-500': page === index + 1,
              })}
              onClick={() => setPage(index + 1)}
              key={_}
            >
              {index + 1}
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default Movies;
