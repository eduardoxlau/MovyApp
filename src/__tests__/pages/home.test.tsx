/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import wait from 'waait';
import { createMemoryHistory } from 'history';
import { MockedProvider } from '@apollo/client/testing';
import { BrowserRouter, Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';

import movies from '__mocks__/movies.mock';
import { GET_LISTS, GET_MOVIES, GET_MOVIES_SEEN } from 'graphql/queries';
import Home from 'views/home';

const mocks = [
  {
    request: {
      query: GET_LISTS,
    },
    result: { data: { getlists: [] } },
  },
  {
    request: {
      query: GET_MOVIES,
    },
    result: { data: { getMovies: { items: movies } } },
  },
  {
    request: {
      query: GET_MOVIES_SEEN,
    },
    result: { data: { getMoviesSeen: movies } },
  },
];

const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

describe('<Home />', () => {
  const history = createMemoryHistory();
  it('should render snapshot Home', async () => {
    const { asFragment } = render(
      <MockedProvider
        mocks={mocks}
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </MockedProvider>
    );
    await wait(0);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should redirect when clicked card', async () => {
    const { getAllByTestId } = render(
      <MockedProvider
        mocks={mocks}
        defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}
      >
        <Router history={history}>
          <Home />
        </Router>
      </MockedProvider>
    );
    await wait(0);
    const card = getAllByTestId('card')[0];
    fireEvent.click(card);
    await wait(200);
    expect(history.length).toBe(2);
    expect(history.location.pathname).toBe(`/trailer/${movies[0].id}`);
  });
});
