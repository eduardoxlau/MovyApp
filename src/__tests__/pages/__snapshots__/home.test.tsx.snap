// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render snapshot Home 1`] = `
<DocumentFragment>
  <div
    class="bg-black flex flex-col text-white"
  >
    <div
      class="w-full relative bg-cover bg-top bg-center flex flex-col"
      style="background-image: url(http://localhost:4000/avatar.png); min-height: 600px;"
    >
      <div
        class="w-full h-full absolute bg-black opacity-50 z-0"
      />
      <div
        class="w-full h-52 bglinear-black-180"
      />
      <div
        class="container flex flex-col md:w-4/5 mx-auto flex-grow z-10"
      >
        <div
          class="py-6 opacity-60"
        >
          2019 . Action, Thiller
        </div>
        <div
          class="flex items-center flex-col md:flex-row"
        >
          <div
            class="text-6xl font-black opacity-80 mb-2"
          >
            International Research Orchestrator
          </div>
          <div
            class="rounded-full border border-white mr-auto md:mr-0 md:ml-auto text-xl px-6 py-1 font-bold opacity-90"
          >
            PG13
          </div>
        </div>
        <div
          class="flex my-8 items-center justify-around md:justify-start"
        >
          <div
            class="bg-pink-600 px-2 py-0.5 rounded md:mr-20 text-sm flex items-center"
          >
            2.30h
          </div>
          <div
            class="mx-10 md:mr-20 text-lg flex items-center text-center flex-col md:flex-row"
          >
            <img
              alt=""
              class="w-6 mr-3"
              src="camera.png"
            />
            <div>
              Quentien cuarentino
            </div>
          </div>
          <div
            class="flex items-center"
          >
            <img
              alt=""
              class="w-4 mr-1"
              src="star.png"
            />
            <img
              alt=""
              class="w-4 mr-1"
              src="star.png"
            />
            <img
              alt=""
              class="w-4 mr-1"
              src="star.png"
            />
            <img
              alt=""
              class="w-4 mr-1"
              src="star.png"
            />
          </div>
        </div>
        <div
          class="mb-40 mt-10 text-lg leading-8"
        >
          Blanditiis similique pariatur provident dolorem repellendus. Accusamus numquam alias in saepe pariatur tenetur. Dolor consequatur quae repudiandae nulla ullam iusto odit repellat qui. Impedit sit facere nostrum nobis dolor. Magnam maiores sunt adipisci excepturi nulla sed aperiam.
        </div>
        <div
          class="flex flex-grow items-center text-lg color-trailer flex-col md:flex-row"
        >
          <div
            class="flex font-thin"
          >
            <div
              class="flex items-center"
            >
              <img
                alt=""
                class="w-5 mr-3"
                src="save.png"
              />
              Watch Later
            </div>
            <div
              class="flex items-center"
            >
              <img
                alt=""
                class="w-5 md:ml-20 mr-3"
                src="play.png"
              />
              <a
                href="/trailer/143"
              >
                Watch Trailer
              </a>
            </div>
            <div
              class="flex items-center"
            >
              <img
                alt=""
                class="w-20 md:ml-20 mr-3"
                src="imdb.png"
              />
            </div>
          </div>
          <div
            class="mt-8 md:mt-0 mx-auto md:mr-0 md:ml-auto"
          >
            <img
              alt=""
              class="w-8"
              src="sound.png"
            />
          </div>
        </div>
      </div>
      <div
        class="w-full h-32 bglinear-black"
      />
    </div>
    <div>
      <div
        class="container text-2xl mx-auto"
      >
        Popular on Movy
      </div>
      <div
        class="flex mt-1 pb-6 relative  w-full container-movies overflow-x-scroll overflow-y-hidden  items-center container-card py-0"
      >
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more2.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more4.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more4.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card2.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container text-2xl mx-auto"
      >
        Continue Watching for undefined
      </div>
      <div
        class="flex mt-1 pb-6 relative  w-full container-movies overflow-x-scroll overflow-y-hidden  items-center container-card py-0"
      >
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more2.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more4.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more4.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card2.png"
              />
            </div>
          </div>
          <div
            class="relative w-full h-1.5 bg-gray-200 rounded"
          >
            <div
              class="absolute h-full bg-blue-500 rounded"
              style="width: 50%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container text-2xl mx-auto"
      >
        Most Viewed
      </div>
      <div
        class="flex mt-1 pb-6 relative  w-full container-movies overflow-x-scroll overflow-y-hidden  items-center container-card p-6"
      >
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view1.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view1.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view1.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view4.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view2.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view3.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view3.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
            >
              <img
                alt=""
                src="http://localhost:4000/view2.png"
              />
              <div
                class="absolute bottom-20 w-full flex justify-center"
              >
                <div
                  class="bg-blue-600 px-4 font-bold text-sm"
                >
                  MOVIE ORIGINAL
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="container text-2xl mx-auto"
      >
        Recommended movies
      </div>
      <div
        class="flex mt-1 pb-6 relative  w-full container-movies overflow-x-scroll overflow-y-hidden  items-center container-card py-0"
      >
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card border-4"
                src="http://localhost:4000/card1.png"
              />
              <img
                alt=""
                class="image-selected w-10 absolute bottom-2"
                data-testid="border-active"
                src="triangle.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card1.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more2.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more4.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/more4.png"
              />
            </div>
          </div>
        </div>
        <div
          class="mx-1 z-0"
        >
          <div
            data-testid="card"
          >
            <div
              class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer"
            >
              <img
                alt=""
                class="img-card"
                src="http://localhost:4000/card2.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full relative bg-cover md:bg-contain bg-no-repeat bg-center md:bg-right bg-center flex flex-col"
      style="background-image: url(http://localhost:4000/avatar.png); min-height: 700px;"
    >
      <div
        class="w-full h-20 bglinear-black-180"
      />
      <div
        class="z-0 w-full h-full absolute bglinear-black-left"
      />
      <div
        class="z-10 mt-20 flex justify-around md:justify-center  text-xl font-bold"
      >
        <div
          class="md:mx-8 pb-1 "
        >
          <a
            href="/home/"
          >
            OVERVIEW
          </a>
        </div>
        <div
          class="md:mx-8 pb-1"
        >
          <a
            href="/home/more"
          >
            MORE LIKE THIS
          </a>
        </div>
        <div
          class="md:mx-8 pb-1"
        >
          <a
            href="/home/details"
          >
            DETAILS
          </a>
        </div>
      </div>
      <div
        class="w-full h-20 bglinear-black flex flex-grow"
      />
    </div>
  </div>
</DocumentFragment>
`;
